context/src/main/java/io/grpc/Context.java

```diff
@@ -946,9 +954,14 @@ public void close() {
    * Key for indexing values stored in a context.
```

Comment:
Rather than adding the parameter-less `#key()` API above, it might be more helpful to avoid confusion around whether the name of a key "matters" (as in https://github.com/grpc/grpc-java/issues/7024) if the Javadoc for `Context.Key` specifies here as well that `Key` values will only be compared using by reference equality, which otherwise is only specified on the `Context#key` creation methods. 

@ejona86 What do you think?

Reply:
We can't really be reviewing code that hasn't had the CLA signed, but I'm fine with adding documentation to `Context.Key`. I'm not sure it will help _much_, but it doesn't hurt.

What looks broken is that we are still saying 'name' in `Context.key(String name)`. That should be `Context.key(String debugName)`. Do you think that'd help?

Reply:
Actually, `debugString` to match [`Attributes.Key`](https://grpc.github.io/grpc-java/javadoc/io/grpc/Attributes.Key.html).

Reply:
See also https://github.com/grpc/grpc-java/issues/1764#issuecomment-348630514

Reply:
Thank you for fast and helpful your feedback.

I also think it is more safe and useful to change parameter name rather than to add parameter-less `Context#key()`.

I updated my code. I think it will clarify the purpose of the parameter of `#key(debugString)`.

Please review the new change. Thank you.